Code/MinimalLib/common.h

```diff
@@ -0,0 +1,309 @@
+    }
+    height = doc["height"].GetUint();
+  }
+
+  if (doc.HasMember("offsetx")) {
+    if (!doc["offsetx"].IsInt()) {
+      return "JSON contains 'offsetx' field, but it is not an int";
+    }
+    offsetx = doc["offsetx"].GetInt();
+  }
+
+  if (doc.HasMember("offsety")) {
+    if (!doc["offsety"].IsInt()) {
+      return "JSON contains 'offsety' field, but it is not an int";
+    }
+    offsety = doc["offsety"].GetInt();
+  }
+
+  if (doc.HasMember("legend")) {
+    if (!doc["legend"].IsString()) {
+      return "JSON contains 'legend' field, but it is not a string";
+    }
+    legend = doc["legend"].GetString();
+  }
+
+  return "";
+}
+
+void get_sss_json(const ROMol &d_mol, const ROMol &q_mol,
+                  const MatchVectType &match, rj::Value &obj,
+                  rj::Document &doc) {
+  rj::Value rjAtoms(rj::kArrayType);
+  for (const auto &pr : match) {
+    rjAtoms.PushBack(pr.second, doc.GetAllocator());
+  }
+  obj.AddMember("atoms", rjAtoms, doc.GetAllocator());
+
+  rj::Value rjBonds(rj::kArrayType);
+  for (const auto qbond : q_mol.bonds()) {
+    unsigned int beginIdx = qbond->getBeginAtomIdx();
+    unsigned int endIdx = qbond->getEndAtomIdx();
+    if (beginIdx >= match.size() || endIdx >= match.size()) {
+      continue;
+    }
+    unsigned int idx1 = match[beginIdx].second;
+    unsigned int idx2 = match[endIdx].second;
+    const auto bond = d_mol.getBondBetweenAtoms(idx1, idx2);
+    if (bond != nullptr) {
+      rjBonds.PushBack(bond->getIdx(), doc.GetAllocator());
+    }
+  }
+  obj.AddMember("bonds", rjBonds, doc.GetAllocator());
+}
+
+std::string mol_to_svg(const ROMol &m, unsigned int w, unsigned int h,
+                       const std::string &details = "") {
+  std::vector<int> atomIds;
+  std::vector<int> bondIds;
+  std::string legend = "";
+  int offsetx = 0, offsety = 0;
+  if (!details.empty()) {
+    auto problems = process_details(details, w, h, offsetx, offsety, legend,
+                                    atomIds, bondIds);
+    if (!problems.empty()) {
+      return problems;
+    }
+  }
+
+  MolDraw2DSVG drawer(w, h);
+  if (!details.empty()) {
+    MolDraw2DUtils::updateDrawerParamsFromJSON(drawer, details);
+  }
+  drawer.setOffset(offsetx, offsety);
+
+  MolDraw2DUtils::prepareAndDrawMolecule(drawer, m, legend, &atomIds, &bondIds);
+  drawer.finishDrawing();
+
+  return drawer.getDrawingText();
+}
+
+std::string get_descriptors(const ROMol &m) {
+  rj::Document doc;
+  doc.SetObject();
+
+  Descriptors::Properties props;
+  std::vector<std::string> dns = props.getPropertyNames();
+  std::vector<double> dvs = props.computeProperties(m);
+  for (size_t i = 0; i < dns.size(); ++i) {
+    rj::Value v(dvs[i]);
+    const auto srt = rj::StringRef(dns[i].c_str());
+    doc.AddMember(srt, v, doc.GetAllocator());
+  }
+
+  if (std::find(dns.begin(), dns.end(), std::string("amw")) == dns.end()) {
+    rj::Value v(Descriptors::calcAMW(m));
+    doc.AddMember("amw", v, doc.GetAllocator());
+  }

```

Comment:
Interesting. I assume "amw" does not get calculated by `props.computeProperties(m)`, and we need to get it separately. And also that it is the only descriptor for which we need to do that, right?

Reply:
This piece of code no longer needs to be in there since I added amw to the property interface elsewhere in the PR.
It's now removed.